nature/static/nature/js/NatureOLMapWidget.js

```diff
@@ -126,15 +129,20 @@ ol.inherits(GeometryTypeControl, ol.control.Control);
             });
             if (self.ready) {
                 self.serializeFeatures();
-                if (!self.options.is_collection) {
+                if (self.isDrawingMultiGeometry()) {
+                    // Prevent switching geometry types but allow adding same type geometries
+                    self.hideGeomTypeIcons();
+                } else if (!self.options.is_collection) {
                     self.disableDrawing(); // Only allow one feature at a time
                 }
             }
         });
 
         var initial_value = document.getElementById(this.options.id).value;
+        var initialGeomType = '';
         if (initial_value) {
             var features = jsonFormat.readFeatures('{"type": "Feature", "geometry": ' + initial_value + '}');
+            initialGeomType = features[0].getGeometry().getType();
```

Comment:
Are we sure that feature[0] always exist here? It is assumed later down in the code that this is an array, will that always be the case?